extension/services/ExtensionMetadataReader.cfc

```diff
@@ -0,0 +1,327 @@
+component accessors=true {
+
+	property name="s3root"            type="string" default="";
+	property name="extensionMeta"     type="query";
+	property name="extensionVersions" type="struct";
+
+	variables._simpleCache = {};
+
+	function loadMeta() {
+		var meta           = _readExistingMetaFileFromS3();
```

Comment:
We had a lot of problems with concurrency previously, might need some locking?

the logic was to always fall back on the cached extensions metadata from s3, if it was updating, i.e couldn't immediately get a lock